### DISCUSSION TITLE
How the tokenizer works?

### DISCUSSION THREAD

**Discussion by S1M0N38** [olimorris/codecompanion.nvim#254] [Q&A] [answered]
I was reviewing the function [`calculate`](https://github.com/olimorris/codecompanion.nvim/blob/2b649f0d10a04cbbf45c5cc555ce0db8f8743293/lua/codecompanion/utils/tokens.lua#L9-L34) in `lua/codecompanion/utils/tokens.lua` to understand how tokens are computed.

It seems that in this implementation, a "word" is treated as 1 token. However, this differs from how LLMs typically generate tokens from text.

Can someone elaborate on this implementation of token counting?

**Comment by olimorris** [comment]
Yep that's correct if I recall. I implemented it as a baseline at the very start of the plugin before I returned the token count from the APIs themselves.

**Reply by S1M0N38** [reply]
so it is just used to estimate the number of tokens sent to LLM and block the request if too many tokens are sent?

**Reply by olimorris** [reply, marked as answer]
I think it's only used in your help slash command and in the Anthropic adapter to crudely determine if a prompt is above a threshold (for prompt caching purposes)
